client/src/app/site/pages/meetings/pages/participants/pages/participant-list/components/participant-list/participant-list.component.ts

```diff
@@ -186,15 +188,29 @@ export class ParticipantListComponent extends BaseMeetingListViewComponent<ViewU
             vote_delegated_to_id: user.vote_delegated_to_meeting_user_id()
         });
 
-        dialogRef.afterClosed().subscribe(result => {
+        dialogRef.afterClosed().subscribe(async result => {
             if (result) {
                 if (!result.group_ids?.length) {
                     result.group_ids = [this.activeMeeting!.default_group_id];
                 }
                 if (result.vote_delegated_to_id === 0) {
                     result.vote_delegated_to_id = null;
                 }
-                this.repo.update(result, user).resolve();
+                const title = this.translate.instant(`This action will remove you from one or more groups.`);
+                const content = `This may diminish your ability to do things in this meeting and you may not be able to revert it by youself. Are you sure you want to do this?`;
+                if (
+                    !(
+                        user.id === this.operator.operatorId &&
+                        this.operator.user
+                            .group_ids()
+                            .filter(group => group !== this.activeMeeting.default_group_id)
+                            .some(id => !(result.group_ids ?? []).includes(id)) &&
+                        !result.group_ids.includes(this.activeMeeting.admin_group_id)
+                    ) ||
+                    (await this.prompt.open(title, content))
+                ) {
+                    this.repo.update(result, user).resolve();
+                }
```

Comment:
This seems quite duplicated to the code in `participant-detail-view.component.ts`. Could this be done elsewhere? Maybe add an `isMeetingAdmin` method to the operator or active meeting service.